core/dotnet2.2/QUICKSTART.md

```diff
@@ -79,17 +115,27 @@ zip -r -0 helloDotNet.zip *
 
 You need to specify the name of the function handler using `--main` argument.
 The value for `main` needs to be in the following format:
-`{Assembly}::{Class Full Name}::{Method}`, e.q.,
-`Apache.OpenWhisk.Example.Dotnet::Apache.OpenWhisk.Example.Dotnet.Hello::Main`
+`{Assembly}::{Class Full Name}::{Method}`, e.q.:
+
++ Synchronous: `Apache.OpenWhisk.Example.Dotnet::Apache.OpenWhisk.Example.Dotnet.Hello::Main`
++ Asynchronous: `Apache.OpenWhisk.Example.Dotnet::Apache.OpenWhisk.Example.Dotnet.Hello::MainAsync`
```

Comment:
An example of the user having to make a decision where they technically don't need to, adding friction.

Reply:
We could simplify the docs - to show the sync case with `Main` first because i think it's simpler.
Then describe the async case. The function can be called `Main` in both since the name isn't material.
I think you're right in that as shown it could lead one to think the name/`Async` suffix is semantically meaningful when it's not.

Reply:
Glad to hear you agree. After looking again at @shawnallen85 's changs though, I agree there is one spot it's important to show both. That's at the very top where it says "that has the exact signature as follows", because in the async case, the signature is materially different. I think to keep it simple, everywhere after that first part, it should use just `Main` for the function name, and show one example regardless of sync/async, because the user would deploy the sync/async versions (if the main function were named `Main`) the exact same way.

Reply:
Do you want to take a shot at editing the quick start along these lines?
We do need to show the `async` case for sure, I've worked with enterprise devs who have asked how to write async examples.

Reply:
Sure, this PR is already merged in, so I'll do a new PR when I have time (most of my OpenWhisking is done on the weekends right now) with suggested doc changes and ask for a review. I think for this PR, we're all set.